src/Pact/Types/PactValue.hs

```diff
@@ -89,3 +107,216 @@ toPactValueLenient t = case toPactValue t of
+
+instance SizeOf PactGuard where
+  sizeOf (PactGuard pid pn) =
+    (constructorCost 2) + (sizeOf pid) + (sizeOf pn)
+
+instance SizeOf KeySet where
+  sizeOf (KeySet pkArr ksPred) =
+    (constructorCost 2) + (sizeOf pkArr) + (sizeOf ksPred)
+
+instance SizeOf KeySetName where
+  -- newtype is free
+  sizeOf (KeySetName n) = sizeOf n
+
+instance SizeOf ModuleGuard where
+  sizeOf (ModuleGuard md n) =
+    (constructorCost 2) + (sizeOf md) + (sizeOf n)
+
+instance (SizeOf p) => SizeOf (UserGuard p) where
+  sizeOf (UserGuard n arr) =
+    (constructorCost 2) + (sizeOf n) + (sizeOf arr)
+
+instance SizeOf Name where
+  sizeOf (QName qn) = (constructorCost 1) + sizeOf qn
+  sizeOf (Name bn) = (constructorCost 1) + sizeOf bn
+
+instance SizeOf QualifiedName where
+  sizeOf (QualifiedName modName n i) =
+    (constructorCost 3) + (sizeOf modName) + (sizeOf n) + (sizeOf i)
+
+instance SizeOf BareName where
+  sizeOf (BareName n i) =
+    (constructorCost 2) + (sizeOf n) + (sizeOf i)
+
+instance SizeOf ModuleName where
+  sizeOf (ModuleName mn namespace) =
+    (constructorCost 2) + (sizeOf mn) + (sizeOf namespace)
+
+instance SizeOf Info where
+  sizeOf (Info may) = sizeOf may
+
+instance SizeOf Parsed where
+  sizeOf (Parsed d len) =
+    (constructorCost 2) + (sizeOf d) + (sizeOf len)
+
+instance SizeOf Delta where
+  -- Most of delta's arguments are unpacked (unboxed) Int64
+  -- Most unboxed types just take one word
+  sizeOf (Columns _ _) = 2 * wordSize
+  sizeOf (Tab _ _ _) = 3 * wordSize
+  sizeOf (Lines _ _ _ _) = 4 * wordSize
+  -- bs is the only packed argument
+  sizeOf (Directed bs _ _ _ _) =
+    (constructorCost 1) + (4 * wordSize) + (sizeOf bs)   
+
+instance SizeOf Code where
+  -- newtype is free
+  sizeOf (Code c) = sizeOf c
+
+instance (SizeOf a, SizeOf b) => SizeOf (a,b) where
+  sizeOf (a,b) = (constructorCost 3) + (sizeOf a) + (sizeOf b)
+
+instance (SizeOf a) => SizeOf (Maybe a) where
+  sizeOf (Just e) = (constructorCost 1) + (sizeOf e)
+  sizeOf Nothing = constructorCost 0
+
+instance SizeOf NamespaceName where
+  -- newtype is free
+  sizeOf (NamespaceName n) = sizeOf n
+
+instance (SizeOf a) => SizeOf [a] where
+  sizeOf arr = arrSize
+    where
+      arrSize = ((1 + (3 * arrLength)) * wordSize) + sizeOfContents
+      arrLength = fromIntegral (L.length arr)
+      sizeOfContents = L.foldl' (\acc e -> acc + (sizeOf e)) 0 arr
+
+instance SizeOf PublicKey where
+  -- newtype is free
+  sizeOf (PublicKey bs) = sizeOf bs
+
+instance SizeOf BS.ByteString where
+  sizeOf bs = byteStringSize
+    where
+      byteStringSize = (9 * wordSize) + byteStringLength
+      byteStringLength = fromIntegral (BS.length bs)
+
+instance SizeOf PactId where
+  -- newtype is free
+  sizeOf (PactId t) = sizeOf t
+
+instance SizeOf FieldKey where
+  -- newtype is free
+  sizeOf (FieldKey t) = sizeOf t
+
+instance SizeOf Text where
+  sizeOf t = (6 * wordSize) + (2 * (fromIntegral (T.length t)))
+
+instance SizeOf BigNat where
```

Comment:
Where are we using this?

Reply:
We're using it when sizing `Integer` (https://github.com/kadena-io/pact/pull/676/files#diff-e7679ea20830b8fa20610f2fdf9ac01bR292-R293) using the `GHC.Integer.GMP.Internals` library from `integer-gmp`.

Reply:
latest commit gets rid of this